illumina2vcf/vcf.py

```diff
@@ -25,11 +25,41 @@ class VCFMaker:
     def __init__(self, genome_reader: ReferenceGenome, indel_records: Dict = {}):
         self._genome_reader = genome_reader
         self._indel_records = indel_records
+        self.qc_stats = {
+            "vcf_lines_autosomal": 0,
+            "called_lines_autosomal": 0,
+            "heterozygous_lines_autosomal": 0,
+            "called_lines_x": 0,
+            "heterozygous_lines_x": 0,
+            "vcf_lines_y": 0,
+            "called_lines_y": 0,
+        }
```

Comment:
🔧 rather than a dict, for internal storage a series of private variables is better. i.e.

```python
class VCFMaker:
    _genome_reader: ReferenceGenome
    _indel_records: Dict
    _qc_lines_autosomal_vcf: int = 0
    _qc_lines_autosomal_called: int = 0
    _qc_lines_autosomal_heterozygous: int = 0
    _qc_lines_x_called: int = 0
    _qc_lines_x_heterozygous: int = 0
    _qc_lines_y_vcf: int = 0
    _qc_lines_y_called: int = 0
    
    def __init__(self, genome_reader: ReferenceGenome, indel_records: Dict = {}):
        self._genome_reader = genome_reader
        self._indel_records = indel_records
```

and then updating the rest of the references to match.

Reply:
Could also have a separate QCresult dataclass, or even a custom class that is also responsible for the calculations. A custom class would also mean that the calculations can be more easily tested, without needing the rest of the pieces.